prime-router/docs/design/validation/design.md

```diff
@@ -0,0 +1,286 @@
+# Problem Statement
+As a sender I want submitted data to be validated to ensure data integrity and quality. As a receiver I want received
+data to be validated to ensure data integrity and quality. For this to happen, there needs to exist a validation 
+setting as well as reusable validation components to perform the sending and receiving validation. 
+
+## Validation Design
+The diagram below proposes four validation "checkpoints":
+
+| Name                      | Description + Location                                              | Purpose                                                                                                        |
+|---------------------------|---------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------|
+| Submission                | Upon receipt of a submission in the SenderReceiver                  | Ensure data is parseable (i.e. not a png or other garbage data)                                                |
+| Pre-transform (sender)    | Per bundle after de-batch in the convert function                   | Ensure data is ready for transformation (i.e. satisfies the validation profile configured for this sender)     |
+| Post-transform (sender)   | After sender transforms and enrichment in the convert function      | Ensure data is valid after transformation (i.e. satisfies the validation profile configured for this sender)   | 
+| Post-transform (receiver) | After receiver translation and enrichment in the translate function | Ensure data is valid after transformation (i.e. satisfies the validation profile configured for this receiver) | 
+
+This is where they are located in the overall system:  
+TODO: update this diagram
+![annotated-fhir-architecture-diagram.png](annotated-fhir-architecture-diagram.png)
+
+A factory/builder shall consume the validationProfile setting and produce reusable, cached validation objects 
```

Comment:
Rewrite to not be FHIR specific, something like: 

A factory/builder shall consume the validationProfile setting and produce reusable, cached validation objects for a particular data format (i.e. in the case of FHIR, instance of configured FhirValidator in a wrapper/interface) that can be used to validate the data.